client/src/App.js

```diff
@@ -352,13 +353,19 @@ function App() {
   }
 
   function showArrow(direction){
+    log.debug("show arrow:", direction);
     expect(direction).oneOf(["north", "south", "west", "east"]);
     setArrow({
       direction,
     });
   }
 
   function hideArrow(){
+    log.debug("hide arrow");
+    //to avoid useless refresh of this component, check current arrow object
+    if(arrow.direction === undefined){
+      return;
+    }
     setArrow({
     });
```

Comment:
Before, without the `if` above, the app would refresh/render the component every time when moving the map by mouse.